test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/UndefOrTest.scala

```diff
@@ -8,188 +8,187 @@
-      expect(none orElse "yes").toEqual("yes")
+  @Test def collect_should_call_guard_at_most_once(): Unit = {
+    var witness = 0
+    def guard(x: String): Boolean = {
+      witness += 1
+      true
     }
+    assertEquals("ok", some("hello") collect {
+      case x @ "hello" if guard(x) => "ok"
+    })
+    assertEquals(1, witness)
+  }
 
-    it("toList") {
-      import scala.scalajs.js.JSConverters._
+  @Test def orElse(): Unit = {
+    assertTrue((some(true) orElse some(false)).get)
+    assertEquals("ok", some("ok") orElse none)
+    assertEquals("yes", none orElse some("yes"))
+    assertJSUndefined(none orElse none)
 
-      expect(some("hello").toList.toJSArray).toEqual(js.Array("hello"))
-      expect(none[String].toList.toJSArray).toEqual(js.Array())
-    }
+    // #2095
+    assertEquals("ok", some("ok") orElse "yes")
+    assertEquals("yes", none orElse "yes")
+  }
 
-    it("toLeft and toRight") {
-      expect(some("left").toLeft("right").isInstanceOf[Left[_, _]]).toBeTruthy
-      expect(none[String].toLeft("right").isInstanceOf[Right[_, _]]).toBeTruthy
-      expect(some("right").toRight("left").isInstanceOf[Right[_, _]]).toBeTruthy
-      expect(none[String].toRight("left").isInstanceOf[Left[_, _]]).toBeTruthy
-    }
+  @Test def toList(): Unit = {
+    import scala.scalajs.js.JSConverters._
 
-    it("toOption") {
-      expect(some("foo").toOption == Some("foo")).toBeTruthy
-      expect(none.toOption == None).toBeTruthy
-    }
+    assertEquals(List("hello"), some("hello").toList)
+    assertEquals(List.empty[String], none[String].toList)
+  }
 
+  @Test def toLeft_and_toRight(): Unit = {
+    assertTrue(some("left").toLeft("right").isInstanceOf[Left[_, _]])
+    assertTrue(none[String].toLeft("right").isInstanceOf[Right[_, _]])
+    assertTrue(some("right").toRight("left").isInstanceOf[Right[_, _]])
+    assertTrue(none[String].toRight("left").isInstanceOf[Left[_, _]])
   }
 
-  describe("scala.scalajs.js.JSConverters.JSRichOption") {
+  @Test def toOption(): Unit = {
+    assertTrue(some("foo").toOption == Some("foo"))
+    assertTrue(none.toOption == None)
+  }
 
-    import js.JSConverters._
+  // scala.scalajs.js.JSConverters.JSRichOption
 
-    it("should provide orUndefined") {
-      expect(Some("asdf").orUndefined).toEqual("asdf")
-      expect((None: Option[String]).orUndefined).toBeUndefined
+  import js.JSConverters._
 
-      // This doesn't work on 2.10, since it doesn't infer
-      // Nothing <:< js.Any to implicitly convert UndefOr[Nothing] to
-      // js.Any
-      // expect(None.orUndefined).toBeUndefined
-    }
+  @Test def should_provide_orUndefined(): Unit = {
+    assertEquals("asdf", Some("asdf").orUndefined)
+    assertJSUndefined((None: Option[String]).orUndefined)
 
+    // This doesn't work on 2.10, since it doesn't infer
+    // Nothing <:< js.Any to implicitly convert UndefOr[Nothing] to
+    // js.Any
+    // assertJSUndefined(None.orUndefined)
```

Comment:
I think this should work with `assertJSUndefined`, because it takes an `Any`, not a `js.Any`. Have you tried?